src/particles/elements/LinearMap.H

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2022-2024 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_ELEMENT_LINEAR_MAP_H
+#define IMPACTX_ELEMENT_LINEAR_MAP_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/alignment.H"
+#include "mixin/beamoptic.H"
+#include "mixin/lineartransport.H"
+#include "mixin/thin.H"
+#include "mixin/named.H"
+#include "mixin/nofinalize.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+
+namespace impactx
+{
+    struct LinearMap
+    : public elements::Named,
+      public elements::BeamOptic<LinearMap>,
+      public elements::Thin,
```

Comment:
@cemitch99 Is this truly always thin? Users can also implement linear drifts in this formalism, don't they?

Reply:
In IMPACT-Z, this element is implemented as a zero-length user-supplied map.  As I see it, the reason to give an element the "thick" attribute is so that it can be _sliced_ - 1) for the application of space charge kicks, and/or 2) for numerical integration if the map is not known analytically, e.g. `RFCavity`.  (Elements like drifts are implemented with a functional dependence on `ds` so that they can be sliced.)  For a user-supplied numerical matrix, it doesn't make sense to slice.

Reply:
That makes sense. What made me wonder: If a user provides the linear transfer matrix of a drift, wouldn't it make potentially sense to slice it (also: it's `.ds()` property is probably ill-defined if we set it to zero?)?

Reply:
Hm.  I think we should distinguish between a "user-defined linear element" and a "user-defined linear map".  What the user originally asked for was the second of these, which could be applied to the beam for many purposes (for example, rotation), and which appears in IMPACT-Z.  This is why I gave it the name `LinearMap`.

To use the term "linear element" suggests that the user has in mind a specific physical model--a  magnet geometry or a set of fields, for example, and a specified thickness `ds`.

Reply:
Looking at your example, though, I do see your point about `ds` and bookkeeping.

Reply:
Updated :)